README.md

```diff
@@ -6,29 +6,26 @@ It contains all the necessary configuration and some placeholder files to get yo
 The best way to create your own project based on this starter is [start.vaadin.com](https://start.vaadin.com/) - you can get only the necessary parts and choose the package naming you want to use.
 
 ## Running the Application
-There are two ways to run the application :  using `mvn spring-boot:run` or by running the `Application` class directly from your IDE.
 
-You can use any IDE of your preference,but we suggest Eclipse or Intellij IDEA.
-Below are the configuration details to start the project using a `spring-boot:run` command. Both Eclipse and Intellij IDEA are covered.
+While developing the application you should always launch the `Application` class in debug mode from your IDE. 
 
-#### Eclipse
-- Right click on a project folder and select `Run As` --> `Maven build..` . After that a configuration window is opened.
-- In the window set the value of the **Goals** field to `spring-boot:run` 
-- You can optionally select `Skip tests` checkbox
-- All the other settings can be left to default
+After the application has started, you can view your it at http://localhost:8080/ in your browser.
 
-Once configurations are set clicking `Run` will start the application
+### Eclipse
 
-#### Intellij IDEA
-- On the right side of the window, select Maven --> Plugins--> `spring-boot` --> `spring-boot:run` goal
-- Optionally, you can disable tests by clicking on a `Skip Tests mode` blue button.
+Open the Application class, right click on it and select "Debug as..." --> "Java Application"
 
-Clicking on the green run button will start the application.
+### Intellij IDEA
 
-After the application has started, you can view your it at http://localhost:8080/ in your browser.
+Open the Application class, right click on it and select "Debug 'Application.main()'"
+
+### Command line
+From the command line you should use the Maven `spring-boot:run` target to run your application. 
+
+You should not launch the Maven `spring-boot:run` target from your IDE as it will prevent breakpoints and debugging from working because it forks a new process. It might also prevent the server from being shut down properly, especially in Eclipse. 
```

Comment:
It would be good to mention that you can debug the application even with `spring-boot:run`, as in the POM file we provide the `jvmArguments` configuration to be able to attach the IDE debugger to the running JVM.